scripts/pca_variant_filter.py

```diff
@@ -74,313 +72,369 @@ def determine_pca_variants(
             "The LD-prune step of this function requires non-preemptible workers only!"
         )
         logger.info("Creating Table after LD pruning of %s...", ld_pruning_dataset)
-        ht = hl.read_table(get_pca_variants_path(ld_pruned=False))
         if ld_pruning_dataset == "ccdg_genomes":
-            vds = hl.vds.read_vds(get_ccdg_vds_path("genomes"))
-            vds = hl.vds.split_multi(vds)
+            vds = get_ccdg_vds("genomes")
+            vds = hl.vds.split_multi(vds, filter_changed_loci=True)
             vds = hl.vds.filter_variants(vds, ht, keep=True)
+            print(vds.variant_data.count_cols())
             mt = hl.vds.to_dense_mt(vds)
         elif ld_pruning_dataset == "gnomad_genomes":
-            mt = get_gnomad_v3_mt(split=True)
-            mt = densify_sites(mt, ht, hl.read_table(gnomad_last_END_position.path))
-            mt = filter_ref_blocks(mt)
+            mt = get_gnomad_v3_mt(key_by_locus_and_alleles=True)
+            logger.info("Converting gnomAD v3.1 MatrixTable to VDS...")
+            mt = mt.select_entries(
+                "END", "LA", "LGT", adj=get_adj_expr(mt.LGT, mt.GQ, mt.DP, mt.LAD)
+            )
+            vds = hl.vds.VariantDataset.from_merged_representation(mt)
+
+            logger.info("Performing split-multi and filtering variants...")
+            vds = hl.vds.split_multi(vds, filter_changed_loci=True)
+            vds = hl.vds.filter_variants(vds, ht)
+
+            logger.info("Densifying data...")
+            mt = hl.vds.to_dense_mt(vds)
         else:
             ValueError(
                 "Only options for LD pruning are `ccdg_genomes` and `gnomad_genomes`"
             )
 
+        hl._set_flags(no_whole_stage_codegen="1")
+        mt = mt.checkpoint(
+            get_pca_variants_path(ld_pruned=False, data=ld_pruning_dataset, mt=True),
+            overwrite=(not read_pre_ld_prune_mt_checkpoint_if_exists),
+            _read_if_exists=read_pre_ld_prune_mt_checkpoint_if_exists,
+        )
+        hl._set_flags(no_whole_stage_codegen=None)
         ht = hl.ld_prune(mt.GT, r2=ld_r2)
         ht = ht.annotate_globals(ld_r2=ld_r2, ld_pruning_dataset=ld_pruning_dataset)
-        ht.checkpoint(
+        ht = ht.checkpoint(
             get_pca_variants_path(ld_pruned=True, data=ld_pruning_dataset),
             overwrite=overwrite,
             _read_if_exists=(not overwrite),
         )
-        mt.filter_rows(hl.is_defined(ht[mt.row_key])).checkpoint(
+        mt = mt.filter_rows(hl.is_defined(ht[mt.row_key]))
+        mt.naive_coalesce(1000).write(
             get_pca_variants_path(ld_pruned=True, data=ld_pruning_dataset, mt=True),
             overwrite=overwrite,
-            _read_if_exists=(not overwrite),
         )
 
 
 def main(args):
-    hl.init(log=f"/variant_filter.log")
+    hl.init(
+        log=f"/variant_filter.log",
+        tmp_dir="gs://ccdg-30day-temp/",
+        default_reference="GRCh38",
+    )
+    hl._set_flags(distributed_scan_comb_op="1")
```

Comment:
What does this flag do?

Reply:
This was used to avoid this error `log4j:ERROR Failed to flush writer, java.io.IOException: No space left on device` earlier when I tried to write a densified MT from VDS. Not sure whether the error is still there, so I am keeping the flag here just in case.

Reply:
Ah, great, can you add a comment basically just `# TODO: This flag can be removed if this error is no longer relevant: log4j:ERROR Failed to flush writer, java.io.IOException: No space left on device when trying to write a densified MT from VDS`